features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/utils/DrawableUtils.kt

```diff
@@ -72,8 +74,52 @@ internal class DrawableUtils(
         return Pair(width, height)
     }
 
-    internal companion object {
-        internal const val MAX_BITMAP_SIZE_IN_BYTES = 10240 // 10kb
+    @Suppress("ReturnCount")
+    private fun getBitmapBySize(
+        displayMetrics: DisplayMetrics,
+        width: Int,
+        height: Int,
+        config: Config
+    ): Bitmap? {
+        var bitmap = getReusableBitmapFromPool(width, height, config)
+        if (bitmap == null) {
+            bitmap = createNewBitmap(displayMetrics, width, height, config)
+        }
+        return bitmap
+    }
+
+    private fun getReusableBitmapFromPool(
+        width: Int,
+        height: Int,
+        config: Config
+    ) = bitmapPool.getBitmapByProperties(width, height, config)
+
+    private fun createNewBitmap(
+        displayMetrics: DisplayMetrics,
+        width: Int,
+        height: Int,
+        config: Config
+    ): Bitmap? {
+        val newBitmap = bitmapWrapper.createBitmap(displayMetrics, width, height, config)
+        newBitmap?.let {
+            bitmapPool.put(it)
+        }
+        return newBitmap
+    }
+
+    private fun registerBitmapPoolForCallbacks(applicationContext: Context) {
```

Comment:
Need to fix issues with pool registration (should be similar to base64serializer logic)